apps/nova/src/components/Chat/Rooms.tsx

```diff
@@ -63,14 +67,23 @@ const Rooms: React.FC<{
             owner={
               room.latestMessage ? room.latestMessage.userId === user.id : false
             }
+            online={
+              usersOnlineMap[room.roomId] &&
+              usersOnlineMap[room.roomId][room.otherMember.id]
+                ? usersOnlineMap[room.roomId][room.otherMember.id]
+                : room.otherMember.online
```

Comment:
This is not clean, we can do better. For example you can move this logic into a function and call it `isOnline`.

```ts
function isOnline(map: UserOnlineMap, userId: number): boolean;
```